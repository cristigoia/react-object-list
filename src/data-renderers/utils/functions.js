/**
 * Gets the value of the cell from the row and keys provided
 * @param  {Object} row  Row supplied
 * @param  {Array|string} keys List of keys used to access the value
 * @return {(Object|string|number)}      Value from row, or row itself if no keys
 */
const getValueFromAccessor = (row, keys) => {
  if (!Array.isArray(keys)) keys = keys.split('.')
  let value = row
  const cloneKeys = [...keys]
  while (value && cloneKeys.length) {
    const thisKey = cloneKeys.shift()
    if (thisKey in value) {
      value = value[thisKey]
    }
  }
  return value
}

export {
  getValueFromAccessor,
}
