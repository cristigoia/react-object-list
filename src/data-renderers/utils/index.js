export {
  getValueFromAccessor,
  handleRowClick,
} from './functions'
