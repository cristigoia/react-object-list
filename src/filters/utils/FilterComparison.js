import React from 'react'
import PropTypes from 'prop-types'
import Select from 'react-select'

class FilterComparison extends React.Component {
  static propTypes = {
    options: PropTypes.arrayOf(PropTypes.shape({
      label: PropTypes.string,
      value: Select.propTypes.value,
    })),
    value: Select.propTypes.value,
    onChange: PropTypes.func.isRequired,
  }

  render() {
    const {options, value} = this.props
    return (
      <div className="objectlist-current-filter__filter-comparison">
        <Select
          options={options}
          value={value}
          onChange={newValue => this.props.onChange(newValue.value)}
          clearable={false}
        />
      </div>
    )
  }
}

export default FilterComparison
