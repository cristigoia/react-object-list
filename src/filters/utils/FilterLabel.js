import React from 'react'
import PropTypes from 'prop-types'

/**
 * Component to render label for filters
 */
class FilterLabel extends React.Component {
  static propTypes = {
    /** Text to render in the label */
    label: PropTypes.string.isRequired,
  }
  render() {
    return (
      <label className="objectlist-current-filter__filter-name">{this.props.label}:</label>
    )
  }
}

export default FilterLabel
