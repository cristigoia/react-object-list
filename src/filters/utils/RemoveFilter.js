import React from 'react'
import PropTypes from 'prop-types'

/**
 * Component to render button used to remove a filter
 */
class RemoveFilter extends React.Component {
  static propTypes = {
    /** Function to be called to remove filter */
    onClick: PropTypes.func.isRequired,
  }

  render() {
    return (
      <button className="apilist-button apilist-button--delete apilist-button--icon apilist-button--borderless" onClick={this.props.onClick} />
    )
  }
}

export default RemoveFilter
