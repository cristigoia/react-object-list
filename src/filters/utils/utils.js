const possibleQueryStringLookupExpressions = ['exact', 'lt', 'gt', 'gte', 'lte']

export {
  possibleQueryStringLookupExpressions,
}
