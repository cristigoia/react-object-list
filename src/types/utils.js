function floatPrecision(floatValue, precision) {
  floatValue = parseFloat(floatValue)
  if (isNaN(floatValue)) { return parseFloat('0').toFixed(precision) } else {
    const power = Math.pow(10, precision)
    floatValue = (Math.round(floatValue * power) / power).toFixed(precision)
    return floatValue.toString()
  }
}

function commaNumber(numberString) {
  const parts = numberString.toString().split('.')
  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',')
  return parts.join('.')
}
/**
 * Converts a float to stringed currency representation
 * @param  {number} floatNumber Amount to convert
 * @return {string}             Dollar string
 */
export function currency(floatNumber) {
  return `${floatNumber < 0 ? '-' : ''}$${commaNumber(floatPrecision(Math.abs(floatNumber), 2))}`
}
