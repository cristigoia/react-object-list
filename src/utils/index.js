export * from './constants'
export * from './proptypes'
export * from './functions'
export loadingSpinner from './loading'
