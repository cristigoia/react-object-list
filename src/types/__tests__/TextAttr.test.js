import React from 'react'
import { snapshotTest } from 'utils/tests'
import { TextAttr } from '../'

describe('TextAttr type', () => {
  it('snapshot tests', () => {
    const input = {data: null}
    const noAttrs = {data: {id: 1, type: 'TestModel'}}
    const allAttrs = {data: {
      id: 1,
      type: 'TestModel',
      attributes: {
        __str__: 'default',
        name: 'abc',
        ref: '123',
        detail: 'some important detail',
      },
    }}
    const testValues = [
      {},
      {value: input},
      {value: noAttrs},
      {value: allAttrs},
      {value: allAttrs, display: 'name'},
      {value: allAttrs, display: 'name', identifier: 'ref'},
      {value: allAttrs, display: 'name', identifier: 'ref', subtext: 'detail'},
    ]
    testValues.forEach(testValue => {
      snapshotTest(<TextAttr {...testValue} />)
    })
  })
})
