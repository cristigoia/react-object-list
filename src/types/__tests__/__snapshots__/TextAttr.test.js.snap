// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAttr type snapshot tests 1`] = `
<span
  className="text-muted"
>
  -
</span>
`;

exports[`TextAttr type snapshot tests 2`] = `
<span
  className="text-muted"
>
  -
</span>
`;

exports[`TextAttr type snapshot tests 3`] = `
<div>
  <div>
    TestModel 1
  </div>
</div>
`;

exports[`TextAttr type snapshot tests 4`] = `
<div>
  <div>
    default
  </div>
</div>
`;

exports[`TextAttr type snapshot tests 5`] = `
<div>
  <div>
    abc
  </div>
</div>
`;

exports[`TextAttr type snapshot tests 6`] = `
<div>
  <div>
    <small
      className="text-muted"
    >
      123
       
    </small>
    abc
  </div>
</div>
`;

exports[`TextAttr type snapshot tests 7`] = `
<div>
  <div>
    <small
      className="text-muted"
    >
      123
       
    </small>
    abc
  </div>
  <small
    className="text-muted"
  >
    some important detail
  </small>
</div>
`;
